### PR TITLE
Power DNS pipeline

### PR DESCRIPTION
**Pull request by andrewstech**
DNSCONTROL

### CODE REVIEW

**Review by wdhdev — CHANGES_REQUESTED** [review]
Can you add support for NS records

**Comment by andrewstech** [comment]
> Can you add support for NS records

done

File: dnsconfig.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  if (domainData.record.NS) {
+    if (Array.isArray(domainData.record.NS)) {
+      for (var ns in domainData.record.NS) {
+        commit[domains[idx].name].push(
+          NS('', domainData.record.NS[ns] + ".")
+        );
+      }
+    } else {
+      commit[domains[idx].name].push(
+        NS('', domainData.record.NS + ".")
```
Comment (wdhdev) [inline dnsconfig.js:98]:
```suggestion
        NS('', `${domainData.record.NS}.`)
```

File: dnsconfig.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      );
+    }
+  }
+
+  // Handle NS records
+  if (domainData.record.NS) {
+    if (Array.isArray(domainData.record.NS)) {
+      for (var ns in domainData.record.NS) {
+        commit[domains[idx].name].push(
+          NS('', domainData.record.NS[ns] + ".")
```
Comment (wdhdev) [inline dnsconfig.js:93]:
```suggestion
          NS('', `${domainData.record.NS}.`)
```

File: dnsconfig.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    commit[domains[idx].name].push(
+      URL('', domainData.record.URL)
+    );
+  }
+
+  // Handle MX records
+  if (domainData.record.MX) {
+    for (var mx in domainData.record.MX) {
+      commit[domains[idx].name].push(
+        MX('', 10, domainData.record.MX[mx] + ".") // Default priority is set to 10
```
Comment (wdhdev) [inline dnsconfig.js:68]:
```suggestion
        MX('', 10, `${domainData.record.MX[mx]}.`) // Default priority is set to 10
```

File: dnsconfig.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      commit[domains[idx].name].push(
+        AAAA('', domainData.record.AAAA[aaaa])
+      );
+    }
+  }
+
+  // Handle CNAME records
+  if (domainData.record.CNAME) {
+    commit[domains[idx].name].push(
+      CNAME('', domainData.record.CNAME + ".")
```
Comment (wdhdev) [inline dnsconfig.js:53]:
```suggestion
      CNAME('', `${domainData.record.CNAME}.`)
```

**Review by wdhdev — CHANGES_REQUESTED** [review]

**Comment by wdhdev** [comment]
Also, `Validation / DNS` checks are failing, is there a reason for this?

**Comment by andrewstech** [comment]
> Also, `Validation / DNS` checks are failing, is there a reason for this?

Probably but it’s 4am I will fix it later

**Review by wdhdev — CHANGES_REQUESTED** [review]
Can we also add support for SRV records, that would be good. If possible set SRV records to be in an object like this:
```js
"SRV": {
  "priority": 10,
  "weight": 1,
  // ...
}
```

Here are the DNSControl docs on it: https://github.com/StackExchange/dnscontrol/blob/main/documentation/language-reference/domain-modifiers/SRV.md

**Comment by EducatedSuddenBucket** [comment]
@wdhdev for support of srv u need to add many underscores to be valid like for eg
For minecraft there is _minecraft._tcp.sub.json

**Pull request closed without merging**